pkg/apis/app.kubermatic/v1/doc.go

```diff
@@ -0,0 +1,20 @@
+/*
+Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// +groupName=app.kubermatic.k8c.io
```

Comment:
In the proposal we have specified app**s** as plural. Was there any reason to change this? In the kubernetes api groups they also use the term apps https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#-strong-api-groups-strong-

Reply:
my mistake

Reply:
@SimonTheLeg  In fact, the content of the field is not a version, but a  `semverlib.Constraints`. it can be (list is not exhaustive):

- a version: `1.2.3`
- a version with wildcard `1.18.*`
- a range of version `>= 1.16.0, < 1.19.15`

AFAIK there is no regex for checking constraints, even so, I doubt that it will be easily understandable...
The SemverConstraints has a validate Validate method that checks if a version comply with the constraint

Moreover, we will have to test that the application is "compatible" with the user cluster in several places:
- in ApplicationInstallerController to not install "incompatible" app
- in KPP API to return a list of compatible applications to the user when he installs an app through the dashboard.
I think validation logic should be in one place and close to the data to validate.


Concerning the crd:

- I have updated the description with these examples to avoid such confusion
- i think the validation should be done in the validating webhook